### ISSUE TITLE
Accessing @conn in Commander?

### ISSUE THREAD

**Issue by guidotripaldi** [grych/drab#110]
I'm wondering which is the canonical way to be able to access `@conn` in a Commander handler. For example, I've tried to render peekable `@conn` adding `<%= if true do %><% conn = @conn %><% end %>` in the template, like in the following example:

```elixir
# lib/commander/page_commander.ex

...

  defhandler peek_test(socket, sender) do
     socket
     |> peek("index.html", :conn)
     |> do_something_with_conn()
  end
```

```elixir
# lib/commander/page_controller.ex

...

  def index(conn, _) do
    render(@conn, "index.html")
  end
```

```elixir
# lib/templates/page/index.html.drab

    <%= if true do %>
    <% conn = @conn %>
    <% end %>

    <button type="button" drab="click: peek_test">Peek Test</button>

```

It works, but is it legitimate, or is there a better way?

**Comment by grych** [comment]
You can always put it in the session and access the session in the commander, see https://tg.pl/drab#store

But the real question is, what do you need conn for? You should not need it after the page is rendered.

**Comment by guidotripaldi** [comment]
It was more a 'philosophical' question than practical, to better understand how Drab to expose assigns. But there was also a practical reason: having access to `@conn` can be useful when you have to call from the Commander some library functions that expect the current `@conn` as a parameter (of course only if they doesn't attempt to modify it)

**Comment by grych** [comment]
Ahh, this is a bug. `@conn` should not be available in this case. Fixing it.

**Comment by grych** [comment]
`@conn` case must be reconsidered.

I am going to use @OvermindDL1's idea from #51 to use only the specific fields of it (configurable).

**Comment by guidotripaldi** [comment]
The reason for that is to reduce the payload size? In this case maybe can be useful allow to get all the structure if wanted/needed, leaving the performances responsibility of this choice on the developer

**Comment by grych** [comment]
This is what we want to do, see #51

**Comment by grych** [comment]
Fixed in v0.7.7
